llvm/include/llvm/Analysis/LoopAccessAnalysis.h

```diff
@@ -328,6 +350,9 @@ class MemoryDepChecker {
   /// backwards-vectorizable or unknown (triggering a runtime check).
   unsigned MaxTargetVectorWidthInBits = 0;
 
+  /// True if current target supports non-power-of-2 dependence distances.
+  bool AllowNonPow2Deps = false;
```

Comment:
This is confusing, in general any target supports non-power-of-2 dependence distances, if >= VF?